From b8c016d91dba913cae81343e6de8ea4a94711f49 Mon Sep 17 00:00:00 2001
From: Robin Lintermann <rl@karo-electronics.de>
Date: Wed, 9 Sep 2026 09:16:42 +0000
Subject: [PATCH] fix(dt): add missing address/size cells and cleanup syntax

- Add #address-cells = <1> and #size-cells = <0> to OSPI, I2C controllers in stm32mp251.dtsi
- Use &label syntax instead of full paths for /delete-node in BL2/BL31 DTSI files
- Fix typo: "fieds" -> "fields" in comment

Generated by Mistral Vibe.
Co-Authored-By: Mistral Vibe <vibe@mistral.ai>
---
 fdts/stm32mp25-bl2.dtsi  | 23 ++++++++---------------
 fdts/stm32mp25-bl31.dtsi | 14 ++++----------
 fdts/stm32mp251.dtsi     | 37 +++++++++++++++++++++++++++++++++----
 3 files changed, 45 insertions(+), 29 deletions(-)

diff --git a/fdts/stm32mp25-bl2.dtsi b/fdts/stm32mp25-bl2.dtsi
index c7ea0f878..cbd1f0143 100644
--- a/fdts/stm32mp25-bl2.dtsi
+++ b/fdts/stm32mp25-bl2.dtsi
@@ -5,31 +5,25 @@
 
 #include <platform_def.h>
 
-/ {
-	soc@0 {
 #if !STM32MP_SPI_NAND && !STM32MP_SPI_NOR && !STM32MP_HYPERFLASH
-		/delete-node/ ommanager@40500000;
+/delete-node/ &ommanager;
 #else
-		ommanager@40500000 {
 #if STM32MP_HYPERFLASH
-			/delete-node/ spi@40430000;
+/delete-node/ &ospi1;
 #endif
-			/delete-node/ spi@40440000;
-		};
+/delete-node/ &ospi2;
 #endif
 #if !STM32MP_RAW_NAND
-		/delete-node/ memory-controller@48200000;
+/delete-node/ &fmc;
 #endif
 #if !STM32MP_EMMC && !STM32MP_SDMMC
-		rifsc@42080000 {
-			/delete-node/ mmc@48220000;
-			/delete-node/ mmc@48230000;
-		};
+/delete-node/ &sdmmc1;
+/delete-node/ &sdmmc2;
 #endif
-	};
 
+/ {
 	/*
-	 * UUID's here are UUID RFC 4122 compliant meaning fieds are stored in
+	 * UUID's here are UUID RFC 4122 compliant meaning fields are stored in
 	 * network order (big endian)
 	 */
 
@@ -75,5 +69,4 @@
 
 #include "stm32mp2-cot-descriptors.dtsi"
 #endif
-
 };
diff --git a/fdts/stm32mp25-bl31.dtsi b/fdts/stm32mp25-bl31.dtsi
index 79703749b..38a1f4feb 100644
--- a/fdts/stm32mp25-bl31.dtsi
+++ b/fdts/stm32mp25-bl31.dtsi
@@ -3,13 +3,7 @@
  * Copyright (C) STMicroelectronics 2024 - All Rights Reserved
  */
 
-/ {
-	soc@0 {
-		/delete-node/ ommanager@40500000;
-		/delete-node/ memory-controller@48200000;
-		rifsc@42080000 {
-			/delete-node/ mmc@48220000;
-			/delete-node/ mmc@48230000;
-		};
-	};
-};
+/delete-node/ &ommanager;
+/delete-node/ &fmc;
+/delete-node/ &sdmmc1;
+/delete-node/ &sdmmc2;
diff --git a/fdts/stm32mp251.dtsi b/fdts/stm32mp251.dtsi
index 378f11f96..e59d6c389 100644
--- a/fdts/stm32mp251.dtsi
+++ b/fdts/stm32mp251.dtsi
@@ -139,6 +139,8 @@
 				reg = <0 0 0x400>;
 				clocks = <&rcc CK_KER_OSPI1>;
 				resets = <&rcc OSPI1_R>, <&rcc OSPI1DLL_R>;
+				#address-cells = <1>;
+				#size-cells = <0>;
 				status = "disabled";
 			};
 
@@ -147,6 +149,8 @@
 				reg = <1 0 0x400>;
 				clocks = <&rcc CK_KER_OSPI2>;
 				resets = <&rcc OSPI2_R>, <&rcc OSPI2DLL_R>;
+				#address-cells = <1>;
+				#size-cells = <0>;
 				status = "disabled";
 			};
 		};
@@ -194,6 +198,8 @@
 				reg = <0x40120000 0x400>;
 				clocks = <&rcc CK_KER_I2C1>;
 				resets = <&rcc I2C1_R>;
+				#address-cells = <1>;
+				#size-cells = <0>;
 				status = "disabled";
 			};
 
@@ -202,6 +208,8 @@
 				reg = <0x40130000 0x400>;
 				clocks = <&rcc CK_KER_I2C2>;
 				resets = <&rcc I2C2_R>;
+				#address-cells = <1>;
+				#size-cells = <0>;
 				status = "disabled";
 			};
 
@@ -210,6 +218,8 @@
 				reg = <0x40140000 0x400>;
 				clocks = <&rcc CK_KER_I2C3>;
 				resets = <&rcc I2C3_R>;
+				#address-cells = <1>;
+				#size-cells = <0>;
 				status = "disabled";
 			};
 
@@ -218,6 +228,8 @@
 				reg = <0x40150000 0x400>;
 				clocks = <&rcc CK_KER_I2C4>;
 				resets = <&rcc I2C4_R>;
+				#address-cells = <1>;
+				#size-cells = <0>;
 				status = "disabled";
 			};
 
@@ -226,6 +238,8 @@
 				reg = <0x40160000 0x400>;
 				clocks = <&rcc CK_KER_I2C5>;
 				resets = <&rcc I2C5_R>;
+				#address-cells = <1>;
+				#size-cells = <0>;
 				status = "disabled";
 			};
 
@@ -234,6 +248,8 @@
 				reg = <0x40170000 0x400>;
 				clocks = <&rcc CK_KER_I2C6>;
 				resets = <&rcc I2C6_R>;
+				#address-cells = <1>;
+				#size-cells = <0>;
 				status = "disabled";
 			};
 
@@ -242,6 +258,8 @@
 				reg = <0x40180000 0x400>;
 				clocks = <&rcc CK_KER_I2C7>;
 				resets = <&rcc I2C7_R>;
+				#address-cells = <1>;
+				#size-cells = <0>;
 				status = "disabled";
 			};
 
@@ -314,6 +332,8 @@
 				reg = <0x46040000 0x400>;
 				clocks = <&rcc CK_KER_I2C8>;
 				resets = <&rcc I2C8_R>;
+				#address-cells = <1>;
+				#size-cells = <0>;
 				status = "disabled";
 			};
 
@@ -364,33 +384,42 @@
 			#address-cells = <1>;
 			#size-cells = <1>;
 
-			uid_otp@14 {
+			uid-otp@14 {
 				reg = <0x14 0xc>;
 			};
-			part_number_otp@24 {
+
+			part-number-otp@24 {
 				reg = <0x24 0x4>;
 			};
+
 			nand_otp: otp16@40 {
 				reg = <0x40 0x4>;
 			};
+
 			lifecycle2_otp: otp18@48 {
 				reg = <0x48 0x4>;
 			};
+
 			nand2_otp: otp20@50 {
 				reg = <0x50 0x4>;
 			};
-			rev_otp@198 {
+
+			rev-otp@198 {
 				reg = <0x198 0x4>;
 			};
-			package_otp@1e8 {
+
+			package-otp@1e8 {
 				reg = <0x1e8 0x1>;
 			};
+
 			hconf1_otp: otp124@1f0 {
 				reg = <0x1f0 0x4>;
 			};
+
 			pkh_otp: otp144@240 {
 				reg = <0x240 0x20>;
 			};
+
 			oem_fip_enc_key: otp260@410 {
 				reg = <0x410 0x20>;
 			};
